pfio/v2/pathlib.py

```diff
@@ -1,271 +1,756 @@
+            )
+        else:
+            return self.with_segments(
+                self._pure.with_stem(stem)  # type: ignore
+            )
 
-    def _glob_generic(self, pattern):
-        '''Slow and generic implementation of glob.'''
-        base = self.resolve()
-        base_parts = base._parts
-        pattern_parts = (base / pattern)._parts
+    def with_suffix(self: SelfPurePathType, suffix: str) -> SelfPurePathType:
+        return self.with_segments(self._pure.with_suffix(suffix))
 
-        visited_prefixes = set()
-        for p in self._fs.list(base, recursive=True):
-            parent = p
+    def with_segments(
+        self: SelfPurePathType,
+        *args: Union[str, PathLike],
+    ) -> SelfPurePathType:
+        return type(self)(*args, fs=self._fs, scheme=self.scheme)
 
-            while (i := parent.rfind(self.sep)) != -1:
-                parent = parent[:i]
-                if parent in visited_prefixes:
-                    continue
-                visited_prefixes.add(parent)
 
-                target_parts = (base / parent)._parts
+class Path(PurePath):
+    """
+    `pathlib.PosixPath` compatible interface.
 
-                if _test_glob_by_parts(target_parts, pattern_parts):
-                    yield self / parent
+    Args:
+        args: construct paths.
+        fs: target file system.
+        scheme: specify URL scheme. (for `as_uri` method)
 
-            target_parts = base_parts + p.split("/")
+    Note:
+        many methods raise `NotImplementedError`
+        because they require unsupported features of `FS`.
 
-            if _test_glob_by_parts(target_parts, pattern_parts):
-                yield self / p
+        several methods behave slightly different,
+        - `stat` returns `FileStat` object instead of `os.stat_result`.
+        - `glob`, `rglob`, `iterdir` will not return directory type object.
+    """
 
-    def stat(self):
-        return self._fs.stat(self.resolve())
+    def __init__(
+        self,
+        *args: str,
+        fs: FS,
+        scheme: Optional[str] = None,
+    ) -> None:
+        super().__init__(*args, fs=fs, scheme=scheme)
 
-    def unlink(self, missing_ok=False):
-        return self._fs.remove(self.resolve())
+    def _as_relative_to_fs(self) -> str:
+        return _removeprefix(self.as_posix(), self.anchor)
 
-    def rename(self, target):
-        raise NotImplementedError("rename() is unsupported on this system")
+    # ---------------------------------------
+    # pathlib.Path compatible classmethods
+    # ---------------------------------------
 
-    def read_text(self):
-        with self.open() as fp:
-            return fp.read()
+    @classmethod
+    def cwd(cls: Type[SelfPathType]) -> SelfPathType:
+        raise _not_supported()
 
-    def read_bytes(self):
-        with self.open(mode='rb') as fp:
-            return fp.read()
+    @classmethod
+    def home(cls: Type[SelfPathType]) -> SelfPathType:
+        raise _not_supported()
```

Comment:
I agree handling this call as unsupported because pfio.v2.FS doesn't support it, but a naive call leads to this KeyError like in cwd.
```
>>> p.home()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/kuenishi/mm/pfnet/pfio/pfio/v2/pathlib.py", line 377, in home
    raise _not_supported()
          ^^^^^^^^^^^^^^^^
  File "/home/kuenishi/mm/pfnet/pfio/pfio/v2/pathlib.py", line 82, in _not_supported
    cname = stack[1].frame.f_locals["self"].__class__.__name__
            ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^
KeyError: 'self
```

Reply:
I fixed it and added `test_not_implemented` in `test_pathlib.py`.